tutorials/Creating a new Lumigator endpoint.md

```diff
@@ -0,0 +1,581 @@
+the following code to 
+[`backend/api/deps.py`](https://github.com/mozilla-ai/lumigator/blob/main/lumigator/python/mzai/backend/api/deps.py):
+
+```
+def get_task_service(session: DBSessionDep) -> TaskService:
+    task_repo = TaskRepository(session)
+    return TaskService(task_repo)
+
+
+TaskServiceDep = Annotated[TaskService, Depends(get_task_service)]
+```
+
+#### 2.5. Update routes
+
+You are almost there! The last thing you need to do is update the `backend/api/routes/tasks.py`file with new code which
+will map API requests to `TaskService` methods:
+
+```
+from uuid import UUID
+
+from fastapi import APIRouter, status
+
+from mzai.backend.api.deps import TaskServiceDep
+from mzai.schemas.extras import ListingResponse
+from mzai.schemas.tasks import TaskCreate, TaskResponse
+
+router = APIRouter()
+
+
+@router.post("/", status_code=status.HTTP_201_CREATED)
+def create_task(service: TaskServiceDep, request: TaskCreate) -> TaskResponse:
+    return service.create_task(request)
+
+
+@router.get("/{task_id}")
+def get_task(service: TaskServiceDep, task_id: UUID) -> TaskResponse:
+    return service.get_task(task_id)
+
+
+@router.delete("/{task_id}", status_code=status.HTTP_204_NO_CONTENT)
+def delete_task(service: TaskServiceDep, task_id: UUID) -> None:
+    service.delete_task(task_id)
+
+
+@router.get("/")
+def list_tasks(
+    service: TaskServiceDep,
+    skip: int = 0,
+    limit: int = 100,
+) -> ListingResponse[TaskResponse]:
+    return service.list_tasks(skip, limit)
+```
+
+#### 2.6. Test
+
+To test your new endpoint, connect to [http://localhost/docs](http://localhost/docs). You should see a new section like
+the following one:
+
+<p style="text-align: center;">
+<img src="assets/image-20240913-083603.png" alt="List of new methods in the tasks endpoint" >
+List of new methods in the tasks endpoint
+</p>
+
+Here is an example for the creation of a new task:
+
+<p style="text-align: center;">
+<img src="assets/image-20240913-083904.png" alt="New task creation" >
+New task creation
+</p>
+
+And here is the format of a task list (to which we have also added a `summarization_eval` task):
+
+<p style="text-align: center;">
+<img src="assets/image-20240913-084103.png" alt="The current list of tasks on the system" >
+The current list of tasks on the system
+</p>
+
+
+## Conclusions
+
+Creating a new endpoint in Lumigator requires some prior knowledge about the system and due to how the code is
+structured you will have to edit many different files (see the picture below). Once you get the gist of it, though, the
+task can be quite straightforward (albeit a bit repetitive), especially if the endpoint mostly has to deal with DB
+operations.
+
+<p style="text-align: center;">
+<img src="assets/image-20240913-084623.png" alt="The list of new and updated files for the tasks endpoint" >
+The list of new and updated files for the tasks endpoint
+</p>
+
+
+
+You can find the code written for this tutorial in
```

Comment:
will this still be the case in the future?

Reply:
I hope so! I made a branch (not a PR :-)) specifically for this and while I know the code can get stale, I also think that if we change our codebase so much that the tutorial does not make any more sense we might also want to update the reference code. WDYT?

Reply:
sounds good!